File: 2002/101 VB.NET Samples/VB.NET - Framework - Key VB.NET Benefits/Product.vb
'Copyright (C) 2002 Microsoft Corporation
'All rights reserved.
'THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER
'EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES OF
'MERCHANTIBILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.

'Requires the Trial or Release version of Visual Studio .NET Professional (or greater).

Option Strict On
Imports System.Data.SqlClient
Public Class Product

    Private m_strConnectionString As String

#Region " Constants for Database connections "
    ' Database connection constants.
    Protected Const SQL_CONNECTION_STRING As String = _
        "Server=localhost;" & _
        "DataBase=Northwind;" & _
        "Integrated Security=SSPI"

    Protected Const MSDE_CONNECTION_STRING As String = _
        "Server=(local)\NetSDK;" & _
        "DataBase=Northwind;" & _
        "Integrated Security=SSPI"
#End Region

#Region "Constructors"

    ' Default constructor.
    Public Sub New()
        MyBase.New()
        ' Set the default connection string.
        m_strConnectionString = SQL_CONNECTION_STRING
        DoTracing("Product instantiated at " & DateTime.Now.ToLongTimeString)
    End Sub

#End Region

#Region " Methods for managing products "
    ' Simulates a call to a stored procedure to add a supplier to the
    ' Suppliers table.
    Public Sub AddProduct(ByVal ProductName As String, ByVal SupplierID As Integer, ByVal UnitPrice As Decimal)
        '...
        Try
            'cmd = New SqlCommand("AddProduct", cnn)
            '...
            'cmd.ExecuteNonQuery()
            DoTracing("Product.AddProduct called at " & DateTime.Now.ToLongTimeString)
        Catch exp As Exception
            Throw New Exception(exp.Message, exp.InnerException)
        End Try
    End Sub

    ' Simulates a call to a stored procedure to delete a product.
    Public Sub DeleteProduct(ByVal ProductID As Integer)
        '...
        Try
            'cmd = New SqlCommand("DeleteProduct", cnn)
            '...
            'cmd.ExecuteNonQuery()
            DoTracing("Product.DeleteProduct called at " & DateTime.Now.ToLongTimeString)
        Catch exp As Exception
            Throw New Exception(exp.Message, exp.InnerException)
        End Try
    End Sub

    ' Simulates a call to a stored procedure to get a list of all
    ' products in Products table.
    Public Function GetProducts() As DataTable
        '...
        Try
            '...
            'da.Fill(dt)
            DoTracing("Product.GetProducts called at " & DateTime.Now.ToLongTimeString)
            'Return dt
        Catch exp As Exception
            Throw New Exception(exp.Message, exp.InnerException)
        End Try
    End Function

    ' Simulates a call to a stored procedure to update a product.
    Public Sub UpdateProduct(ByVal ProductID As Integer, ByVal ProductName As String, ByVal SupplierID As Integer, ByVal UnitPrice As Decimal)
        '...
        Try
            'cmd = New SqlCommand("UpdateProduct", cnn)
            '...
            'cmd.ExecuteNonQuery()
            DoTracing("Product.UpdateProduct called at " & DateTime.Now.ToLongTimeString)
        Catch exp As Exception
            Throw New Exception(exp.Message, exp.InnerException)
        End Try
    End Sub
#End Region

#Region "Utility functions"
    ' Writes to Application Log.
    Private Sub WriteToLog(ByVal strMsg As String)
        Dim oEventLog As New EventLog("Application")
        oEventLog.Source = "VBNET.HowTo.Contrast VB.NET with VB6"
        oEventLog.WriteEntry(strMsg)
    End Sub

    ' Reports on current object status in various ways.
    Private Sub DoTracing(ByVal strMsg As String)
        ' Create a trace listener for the event log.
        'Dim logTraceListener As New EventLogTraceListener("Product--VB.NET How-To: ContrastVBNETwithVB6")

        ' Add the trace listener to the collection.
        'Trace.Listeners.Add(logTraceListener)

        ' Create a trace listener that will send send trace output to the console
        ' It could have written to a file or stream instead.
        Dim consoleTraceListener As New TextWriterTraceListener(System.Console.Out)

        ' Add the trace listener to the collection.
        Trace.Listeners.Add(consoleTraceListener)

        ' Write output.
        Trace.WriteLine(strMsg)
    End Sub
#End Region

End Class

File: 2003/Code Samples/Framework Samples/Framework - Key Benefits/Product.cs
//Copyright (C) 2002 Microsoft Corporation

//All rights reserved.

//THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER

//EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES OF

//MERCHANTIBILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.

//Requires the Trial or Release version of Visual Studio .NET Professional (or greater).

using System.Data.SqlClient;

public class Product;

    private m_strConnectionstring string;

#region " Constants for Database connections ";

    // Database connection constants.

    protected const SQL_CONNECTION_STRING string = _;

        "Server=localhost;" + _;

        "DataBase=Northwind;" + _;

        "Integrated Security=SSPI";

    protected const MSDE_CONNECTION_STRING string = _;

        "Server=(local)\NetSDK;" + _;

        "DataBase=Northwind;" + _;

        "Integrated Security=SSPI";

#endregion

#region "Constructors";

    // Default constructor.

    public () {

        

        // Set the default connection string.

        m_strConnectionstring = SQL_CONNECTION_STRING;

        DoTracing("Product instantiated at " + DateTime.Now.ToLongTimestring);

    }

#endregion

#region " Methods for managing products ";

    // Simulates a call to a stored procedure to add a supplier to the

    // Suppliers table.

    public void AddProduct(ProductName string, SupplierID int, UnitPrice Decimal);

        //...

        try {

            //cmd = new SqlCommand("AddProduct", cnn)

            //...

            //cmd.ExecuteNonQuery()

            DoTracing("Product.AddProduct called at " + DateTime.Now.ToLongTimestring);

       } catch( exp Exception;

            throw new Exception(exp.Message, exp.InnerException);

        }

    }

    // Simulates a call to a stored procedure to delete a product.

    public void DeleteProduct(ProductID int);

        //...

        try {

            //cmd = new SqlCommand("DeleteProduct", cnn)

            //...

            //cmd.ExecuteNonQuery()

            DoTracing("Product.DeleteProduct called at " + DateTime.Now.ToLongTimestring);

       } catch( exp Exception;

            throw new Exception(exp.Message, exp.InnerException);

        }

    }

    // Simulates a call to a stored procedure to get a list of all

    // products in Products table.

    public DATA_TYPE_HERE GetProducts() DataTable;

        //...

        try {

            //...

            //da.Fill(dt)

            DoTracing("Product.GetProducts called at " + DateTime.Now.ToLongTimestring);

            //return dt

       } catch( exp Exception;

            throw new Exception(exp.Message, exp.InnerException);

        }

    }

    // Simulates a call to a stored procedure to update a product.

    public void UpdateProduct(ProductID int, ProductName string, SupplierID int, UnitPrice Decimal);

        //...

        try {

            //cmd = new SqlCommand("UpdateProduct", cnn)

            //...

            //cmd.ExecuteNonQuery()

            DoTracing("Product.UpdateProduct called at " + DateTime.Now.ToLongTimestring);

       } catch( exp Exception;

            throw new Exception(exp.Message, exp.InnerException);

        }

    }

#endregion

#region "Utility functions";

    // Writes to Application Log.

    private void WriteToLog(strMsg string);

        oEventLog new EventLog("Application");

        oEventLog.Source = "VBNET.HowTo.Contrast C# with VB6";

        oEventLog.WriteEntry(strMsg);

    }

    // Reports on current object status in various ways.

    private void DoTracing(strMsg string);

        // Create a trace listener for the event log.

        //logTraceListener new EventLogTraceListener("Product--C# How-To: ContrastVBNETwithVB6")

        // Add the trace listener to the collection.

        //Trace.Listeners.Add(logTraceListener)

        // Create a trace listener that will send send trace output to the console

        // It could have written to a file or stream instead.

        consoleTraceListener new TextWriterTraceListener(System.Console.Out);

        // Add the trace listener to the collection.

        Trace.Listeners.Add(consoleTraceListener);

        // Write output.

        Trace.WriteLine(strMsg);

    }

#endregion

}

